File: 2478-longest-nice-subarray/2478-longest-nice-subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        int acc = nums[0] ; 
        int ans = 1; 
        int j = 0; 

        for(int i =1; i< nums.size(); i++)
        {
            while((acc & nums[i])!=0 )
            {
                acc = acc ^ nums[j];
                j++;
            }
            acc = acc | nums[i];
            ans = max(ans, i-j+1);
        }return ans;
        
    }
};

File: 2478-longest-nice-subarray/2478-longest-nice-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestNiceSubarray = function(nums) {
    let acc = nums[0] ; 
        let ans = 1; 
        let j = 0; 

        for(let i =1; i< nums.length; i++)
        {
            while((acc & nums[i])!=0 )
            {
                acc = acc ^ nums[j];
                j++;
            }
            acc = acc | nums[i];
            ans = Math.max(ans, i-j+1);
        }return ans;
    
};